### PR TITLE
provider/aws: support tenancy for spot fleets

### LINKED ISSUE #14152: Feature Request: aws_spot_fleet_request placement_tenancy property
Currently the aws_spot_fleet_request resource does not expose the tenancy setting in the launch_specification.

### Terraform Version
v0.9.3

### Affected Resource(s)
aws_spot_fleet_request

### Terraform Configuration Files
```hcl
resource "aws_spot_fleet_request" "fleet" {
  iam_fleet_role              = "${var.spot_fleet_role_arn}"
  spot_price                  = "1"
  target_capacity             = "${var.cluster_capacity}"
  valid_until                 = "2019-10-02T19:33:11Z"
  terminate_instances_with_expiration = true

  launch_specification {
    instance_type        = "c4.xlarge"
    iam_instance_profile = "instance_role"
    ami                  = "${var.ecs_ami}"
    key_name             = "${var.ssh_key_name}"
    availability_zone    = "us-west-2a"

    subnet_id = "${var.cluster_subnet_id}"
    vpc_security_group_ids = ["${var.security_group_id}"]

    ebs_optimized = true

    placement_tenancy = "dedicated"
  }
}
```

### Expected Behavior
Creation of a spot fleet request with dedicated instances.

### Actual Behavior
The following error:
* aws_spot_fleet_request.fleet: launch_specification: invalid or unknown key: placement_tenancy

### Steps to Reproduce
`terraform apply`

### References

[The SpotPlacement object with the tenancy setting
](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_SpotPlacement.html)
[Similar property](https://github.com/hashicorp/terraform/blob/master/builtin/providers/aws/resource_aws_spot_fleet_request.go#L307)

### PR DESCRIPTION
**Pull request by paddycarver**
Add the `placement_tenancy` attribute to the `aws_spot_fleet_request`
resource so users can request instances use dedicated hardware.

Fixes #14152.

### CODE REVIEW

File: builtin/providers/aws/resource_aws_spot_fleet_request_test.go [hunk 1]
```diff
@@ -1251,3 +1296,78 @@ resource "aws_spot_fleet_request" "foo" {
...
+    iam_fleet_role = "${aws_iam_role.test-role.arn}"
+    spot_price = "0.005"
+    target_capacity = 2
+    valid_until = "2019-11-04T20:44:20Z"
+    terminate_instances_with_expiration = true
+    launch_specification {
+        instance_type = "m1.small"
+        ami = "ami-d06a90b0"
+        key_name = "${aws_key_pair.debugging.key_name}"
+	placement_tenancy = "dedicated"
```
Comment (grubernaut) [inline builtin/providers/aws/resource_aws_spot_fleet_request_test.go]: odd spacing here

Comment (paddycarver) [inline builtin/providers/aws/resource_aws_spot_fleet_request_test.go]: Oops, good catch, thanks.

**Comment by paddycarver** [comment]
Re: documentation, I looked into adding documentation with the rest of the `launch_specification` documentation, only to find the `launch_specification` documentation [doesn't list individual properties](https://www.terraform.io/docs/providers/aws/r/spot_fleet_request.html#launch_specification). It just points to the AWS docs about the properties, which document the placement option. Do you have any suggestions on where you'd like to see the docs added? As far as I can tell, I'm following the established pattern, but there's a very good chance I'm just missing where I should be looking.

**Comment by grubernaut** [comment]
@paddycarver :/ odd, assumed it was documented similarly to our other resources. Perhaps just add it to the one of the `launch_specification` examples shown in the documentation? We should look into expanding that whole configuration block into it's own documentation section, but that's out of scope for this pr

**Comment by paddycarver** [comment]
Cool, last push should do that. 👍 Let me know if I misunderstood.

**Pull request merged**